pkg/apiserver/apiserver.go

```diff
@@ -168,6 +170,14 @@ func InstallLogsSupport(mux Mux) {
 	mux.Handle("/logs/", http.StripPrefix("/logs/", http.FileServer(http.Dir("/var/log/"))))
 }
 
+func InstallServiceErrorHandler(container *restful.Container) {
+	container.ServiceErrorHandler(serviceErrorHandler)
+}
+
+func serviceErrorHandler(serviceErr restful.ServiceError, response *restful.Response) {
+	errorJSON(apierrors.NewGenericServerResponse(serviceErr.Code, "", "", "", "", 0, false), latest.Codec, response)
```

Comment:
We'll also want to extract the api version from the path, if possible, falling back to the default version only if the path prefix is totally broken, but it's ok to punt that to the more general issue of sending status using the proper version.